pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -485,46 +465,48 @@ pub mod pallet {
 
 			Ok(())
 		}
-	}
 
-	impl<T: Config> Pallet<T> {
-		pub(crate) fn try_range<'a, D, F>(
-			slice: &mut &'a [u8],
-			next_steps: usize,
-			error: Error<T>,
-			transformer: F,
-		) -> Result<D, DispatchError>
-		where
-			F: Fn(&'a [u8]) -> Result<D, DispatchError>,
-		{
-			ensure!(slice.len() >= next_steps, error);
-
-			let (input, new_slice) = slice.split_at(next_steps);
-			let res = transformer(input)?;
-			*slice = new_slice;
-
-			Ok(res)
+		/// Start packing messages in a single message instead of enqueue
+		/// messages.
+		/// The message will be enqueued once `end_pack_messages()` is called.
+		#[pallet::weight(T::WeightInfo::start_pack_messages())]
+		#[pallet::call_index(9)]
+		pub fn start_pack_messages(origin: OriginFor<T>, destination: Domain) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
```

Comment:
I have a couple of questions to make sure I understand the expected flow.

- Who is expected to call these packing extrinsics? 
  - From what I can tell, in the `OutboundMessageHandler` impl below, it would be a `sender` coming from LP. An LP `sender` can be some address that comes from EVM, right? If so, how does that address manage to call these extrinsics?
- How do we handle cases where the supposed caller errors out between start and end?
  - Should we maybe consider adding a way to manually end a batch for a particular (sender, destination) pair via root?

Reply:
Very good questions!

> Who is expected to call these packing extrinsics?

The same as who call the extrinsics from `pallet-liquidity-pools`. It does not need to be an address from EVM. Just a normal account.

> How do we handle cases where the supposed caller errors out between start and end?

If some call between start and end fails, the user notice that. Note that each call in between is not yet processed by the `on_idle()`, so it fails just in that moment.

As a resume of how all of this work:
```rust
Gateway::start_pack_messages(ALICE)
LiquidityPools::add_pool(ALICE)
LiquidityPools::add_tranche(ALICE) // -> If it fails, ALICE will know it and can choose to try again with the correct params before calling end
Gateway::end_pack_messages(ALICE) // -> Now, the batch message is on the queue to be sent in the `on_idle()` function.
```

The above can be done in several blocks, but you can also batch all of them. If one fails, nothing is applied.

Reply:
OK, thank you for the explanation.

Reply:
Awesome explanation, thanks for that @lemunozm !

IMO, in general we can assume that `{start, end}_packing_messages` will only be called locally from a Substrate account because EVM accounts would call it on their EVM chain of choice such that we receive the LP batch message.